src/Http/Http.Abstractions/src/Extensions/HttpResponseWritingExtensions.cs

```diff
@@ -133,7 +133,7 @@ private static void WriteMultiSegmentEncoded(PipeWriter writer, string text, Enc
             // Therefore, we check encodedLength - totalBytesUsed too.
             while (!completed || encodedLength - totalBytesUsed != 0)
             {
-                encoder.Convert(source, destination, flush: source.Length == 0, out var charsUsed, out var bytesUsed, out completed);
+                encoder.Convert(source, destination, flush: true, out var charsUsed, out var bytesUsed, out completed);
```

Comment:
Should you provide a comment or assert so that future changes won't invalidate the assumption?